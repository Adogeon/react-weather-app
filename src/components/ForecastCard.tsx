import styles from "./ForecastCard.module.css";

interface ForecastCardProps {
  date: string;
  condition: {
    icon: string;
    text: string;
  };
  mintemp_c: string;
  maxtemp_c: string;
  avghumidity: string;
  chance_of_rain: string;
}

const ForecastCard = ({
  date,
  condition,
  mintemp_c,
  maxtemp_c,
  avghumidity,
  chance_of_rain,
}: ForecastCardProps) => {
  return (
    <div className={styles["daily-forecast-card"]}>
      <div>
        <div>{date}</div>
        <div>
          <img src={condition.icon} alt={`${condition.text}-icon`} />
        </div>
        <div>{condition.text}</div>
      </div>
      <div>
        <div>
          {mintemp_c}°C - {maxtemp_c}°C
        </div>
        <div>Humidity: {avghumidity}%</div>
        <div>Chance of Rain: {chance_of_rain}%</div>
      </div>
    </div>
  );
};

export default ForecastCard;
