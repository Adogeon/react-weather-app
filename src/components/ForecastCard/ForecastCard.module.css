.daily-forecast-card {
  flex: 1 0 50%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: 20px;
}

.daily-forecast-card img {
  width: 50px;
  height: 50px;
}

.daily-forecast-card div {
  margin: 0 10px;
}
  