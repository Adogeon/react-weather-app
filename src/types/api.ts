import type { CurrentWeather, DailyForecastWithHour } from "./weather";

export interface rawAPIError {
    error: {
        code: string;
        message: string;
    };
}

export interface rawAPISuccess {
    location: {
        name: string;
        region: string;
        country: string;
    };
    current: CurrentWeather
    forecast: {
        forecastday: Array<DailyForecastWithHour>
    }
};
