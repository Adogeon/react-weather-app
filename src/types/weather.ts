export interface WeatherCondition {
    text: string;
    icon: string;
}

export interface TempAndConditions {
    temp_c: string;
    temp_f: string;
    feelslike_c: string;
    feelslike_f: string;
    condition: WeatherCondition;
}

export interface CurrentWeather extends TempAndConditions {
    last_updated: string
}

interface DailyForecastWeather {
    maxtemp_c: string;
    maxtemp_f: string;
    mintemp_c: string;
    mintemp_f: string;
    daily_chance_of_rain: string;
    avghumidity: string;
    condition: WeatherCondition;
}

export interface DailyForecastData {
    date: string;
    day: DailyForecastWeather
}

interface HourlyForecast extends TempAndConditions {
    time: string;
}

export interface DailyForecastWithHour extends DailyForecastData {
    hour: Array<HourlyForecast>
}