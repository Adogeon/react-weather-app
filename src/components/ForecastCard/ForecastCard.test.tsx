import { render } from "@testing-library/react";
import { test, expect } from "vitest";
import ForecastCard from ".";
import { DailyForecastData } from "../../types/weather";

const testData: DailyForecastData = {
  date: "2024-11-15",
  day: {
    mintemp_c: "12.8",
    mintemp_f: "55",
    maxtemp_c: "14.5",
    maxtemp_f: "58.1",
    avghumidity: "80",
    daily_chance_of_rain: "0",
    condition: {
      text: "test",
      icon: "/64x64/test",
    },
  },
};

test("Forecast Card should render correctly", () => {
  const { asFragment } = render(<ForecastCard data={testData} />);

  expect(asFragment()).toMatchSnapshot();
});
