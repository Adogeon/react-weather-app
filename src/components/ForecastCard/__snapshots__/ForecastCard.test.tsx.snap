// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Forecast Card should render correctly 1`] = `
<DocumentFragment>
  <div
    class="_daily-forecast-card_982c90"
  >
    <div>
      <div>
        2024-11-15
      </div>
      <div>
        <img
          alt="test-icon"
          src="/64x64/test"
        />
      </div>
      <div>
        test
      </div>
    </div>
    <div>
      <div>
        12.8°C - 14.5°C
      </div>
      <div>
        Humidity: 80%
      </div>
      <div>
        Chance of Rain: 0%
      </div>
    </div>
  </div>
</DocumentFragment>
`;
